ping/ping6_common.c

```diff
@@ -832,6 +932,71 @@ int ping6_parse_reply(struct ping_rts *rts, socket_st *sock,
 			fflush(stdout);
 			return 0;
 		}
+	} else if (icmph->icmp6_type == ICMPV6_EXT_ECHO_REPLY) {
+		if (!rts->multicast &&
+		    memcmp(&from->sin6_addr.s6_addr, &rts->whereto6.sin6_addr.s6_addr, 16))
+			return 1;
+		if (!is_ours(rts, sock, icmph->icmp6_id))
+			return 1;
+
+		sequence = ntohs(icmph->icmp6_seq);
+		state = icmph->icmp6_seq & 0xe0;
```

Comment:
You forgot to shift this right by 5, so the switch below won't hit any of the cases.